src/coreclr/jit/emitarm64.cpp

```diff
@@ -6906,7 +6924,58 @@ void emitter::emitIns_R_R_R_I(instruction ins,
     }
     assert(fmt != IF_NONE);
 
-    instrDesc* id = emitNewInstrCns(attr, imm);
+    // An "instrDesc" will *always* be required.
+    // Under normal circumstances the instruction
+    // will be added to the emitted group. However,
+    // this is not correct for instructions that
+    // are going to overwrite already-emitted
+    // instructions and we therefore need space to
+    // hold the new instruction descriptor.
+    instrDesc* id;
+
+    // One cannot simply instantiate an instruction
+    // descriptor, so this array will be used to
+    // hold the instruction being built.
+    unsigned char tempInstrDesc[sizeof(instrDesc)];
```

Comment:
This should use `inlineInstrDesc<instrDesc>` instead.

Reply:
(But, why is it necessary? Can we just set `id = reuseInstr` instead, with similar assert first that I pointed out below?)

Reply:
Hi Jakob. I agree that we can just use **id = reuseInstr** instead. The code now appears as follows:

**assert (sizeof(*reuseInstr) == sizeof(*id));

id = reuseInstr;
memset(id, 0, sizeof(instrDesc));**

I am happy with the "assert", but does it gain anything?

Reply:
No. I think it should be something like:
```cpp
INDEBUG(size_t reusedInstrSize = reuseInstr != nullptr ? emitSizeOfInsDsc(reuseInstr) : 0);
... code that overwrites reuseInstr ...
assert((reuseInstr == nullptr) || (emitSizeOfInsDsc(reuseInstr) == reusedInstrSize));
```

Reply:
As the code no longer allocates space to hold the instruction temporarily, this issue is resolved by default.